pipeline/run_pipeline.py

```diff
@@ -194,10 +203,16 @@ def generate_and_save_candidate_directions(
     return mean_diffs
 
 
-def normalize_vector(vector):
+def normalize_vector(vector,model_base):
     """Normalize a given vector to have unit length."""
-    norm = torch.norm(vector)
-    if norm == 0:
+    if vector.dim() > 1:
+        assert vector.shape == (model_base.model.config.num_hidden_layers, model_base.model.config.hidden_size), "Shape doesnt match"
```

Comment:
I would recommend not checking for this shape. Your `normalize_vector` function should work for vector/matrix of any sizes. So you can normalize other vectors using this function too.

Just the separate check for `vector.dim()` is good. Your function documentation should mention what happens for multi-dimensional tensor inputs (that they are normalized along the last dim)